docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -327,255 +388,102 @@ In Geo-Partitioning, the table must be splitted into partitions, and each partit
-
-An object may contain multiple placement rules, each of which corresponds to a rule in PD.
-
-Advantages of building system table include:
-
-* It’s easy for users to filter and aggregate the result
-* There’s no need to support a new grammar, and it’s easier to implement
-
-### Show placement
-
-But there’re a problem here. The system table only contains stored placement rules, and users cannot query the effective rule of one object from it.
-
-For example, table `t` has two partitions `p0` and `p1`, and a placement rule is added on `t`. If the user wants to query the working rule of `p0`, he will find no placement rule is defined for `p0` through the system table. Based on the rule priorities described later, he must query the placement rule on `t`. This procedure is annoying.
-
-To simplify the procedure, a `SHOW PLACEMENT POLICY` statement is provided to query the effective rule for one specified object.
-
-The statement is in such a format:
-
-```sql
-SHOW PLACEMENT POLICY FOR {DATABASE | SCHEMA} schema_name;
-SHOW PLACEMENT POLICY FOR TABLE table_name [PARTITION partition_name];
-SHOW PLACEMENT POLICY FOR INDEX index_name FROM table_name [PARTITION partition_name];
-```
+### Privilege management
 
-TiDB will automatically find the effective rule based on the rule priorities.
-
-This statement outputs at most 1 line. For example, when querying a table, only the placement rule defined on the table itself is shown, and the partitions and indices in it will not be shown.
-
-The output of this statement contains these fields:
-
-* Target: The object queried. It can be a database, table, partition, or index.
-    * For database, it is shown in the format `DATABASE database_name`
-    * For table, it is shown in the format `TABLE database_name.table_name`
-    * For partition, it is shown in the format `TABLE database_name.table_name PARTITION partition_name`
-    * For index, it is shown in the format `INDEX index_name FROM database_name.table_name`
-* Equivalent placement: A equivalent `ALTER` statement on `target` that defines the placement rule.
-* Existing placement: All the executed `ALTER` statements that affect the placement of `target`, including the statements on its parent.
-* Scheduling state: The scheduling progress from the PD aspect.
-
-### Show create table
+Privilege management is quite straightforward:
 
-It’s useful to show rules in `SHOW CREATE TABLE` statement, because users can check the rules easily.
+* `ALTER [DATABASE|TABLE]` statement requires `Alter` privilege
+* `CREATE TABLE` statement requires `Create` privilege
+* `information_schema.placement_rules` and `SHOW PLACEMENT` only shows the placement rules on the objects that visible to the current user
+* `ADMIN SHOW DDL` requires `Super` privilege
+* `CREATE PLACEMENT POLICY`, `DROP PLACEMENT POLICY` and `ALTER PLACEMENT POLICY` require `PLACEMENT_ADMIN` (a new dynamic privilege). This is because these objects have global scope.
```

Comment:
So a user can set the placement policy directly on their table/partition, but not create/alter a global placement policy? Maybe the namespace should be also per database, so a multi-tenant installation have use for it?

Reply:
I have added this to 'known limitations'.

Multi-tenant systems still work, but tenants can not create or modify the policies. I think this makes sense in practice, since the tenant should not be thinking about the physical provisioning.